projection/ibc_channel/ibc_channel.go

```diff
@@ -324,20 +320,32 @@ func (projection *IBCChannel) HandleEvents(height int64, events []event_entity.E
 				return fmt.Errorf("error updating channel last_activity_time: %w", err)
 			}
 
-			amount := msgIBCAcknowledgement.Params.MaybeFungibleTokenPacketData.Amount.Uint64()
-			denom := msgIBCAcknowledgement.Params.MaybeFungibleTokenPacketData.Denom
-			destinationChannelID := msgIBCAcknowledgement.Params.Packet.DestinationChannel
-			destinationPortID := msgIBCAcknowledgement.Params.Packet.DestinationPort
-
-			if err := projection.updateBondedTokensWhenMsgIBCAcknowledgement(
-				ibcChannelsView,
-				channelID,
-				amount,
-				denom,
-				destinationChannelID,
-				destinationPortID,
-			); err != nil {
-				return fmt.Errorf("error updateChannelBondedTokensWhenMsgIBCAcknowledgement: %w", err)
+			if msgIBCAcknowledgement.Params.MaybeFungibleTokenPacketData.MaybeError == "" {
```

Comment:
Shall we rely on the Nullity of `Result` rather than a empty error string?

Reply:
Thanks! Good catch, I will use `success` field to do the checking here.